### PR TITLE
Use file statuses for checks count.

### PR DESCRIPTION
**Pull request by MancunianSam**
At the moment, we're checking the relevant metadata fields to see if the
checks have completed. The problem is that if they fail, there is no
metadata so the checks page just hangs.

This checks the file statuses instead of the metadat itself because
these will be set to failed if there are any failures.

### CODE REVIEW

File: src/main/scala/uk/gov/nationalarchives/tdr/api/service/ConsignmentService.scala [hunk 1]
```diff
@@ -125,14 +125,6 @@ class ConsignmentService(
     consignmentRepository.consignmentHasFiles(consignmentId)
```
Comment (TomJKing) [inline src/main/scala/uk/gov/nationalarchives/tdr/api/service/ConsignmentService.scala:125]: Looks like there are some unused constructor params now the method is remove, these should be removed as well:

- `fileMetadataRepository: FileMetadataRepository`
- `fileRepository: FileRepository`
- `ffidMetadataRepository: FFIDMetadataRepository`

File: src/main/scala/uk/gov/nationalarchives/tdr/api/service/FileStatusService.scala [hunk 2]
```diff
@@ -25,6 +26,19 @@ class FileStatusService(fileStatusRepository: FileStatusRepository, uuidSource:
     fileStatusRepository.addFileStatuses(rows).map(_.map(row => FileStatus(row.fileid, row.statustype, row.value)).toList)
   }
 
+  def getConsignmentFileProgress(consignmentId: UUID): Future[FileChecks] = {
```
Comment (TomJKing) [inline src/main/scala/uk/gov/nationalarchives/tdr/api/service/FileStatusService.scala:29]: Should add some unit tests for this new function

File: src/main/scala/uk/gov/nationalarchives/tdr/api/service/ConsignmentService.scala [hunk 3]
```diff
@@ -125,14 +125,6 @@ class ConsignmentService(
...
   }
 
-  def getConsignmentFileProgress(consignmentId: UUID): Future[FileChecks] = {
```
Comment (TomJKing) [inline src/main/scala/uk/gov/nationalarchives/tdr/api/service/ConsignmentService.scala:128]: I think the following functions can now be deleted, as they don't appear to be used anywhere expect in their own unit tests, and the unit tests can be removed as well:

- `def countProcessedAvMetadataInConsignment(consignmentId: UUID): Future[Int]`
- `def countProcessedChecksumInConsignment(consignmentId: UUID): Future[Int]`
- `def countProcessedFfidMetadata(consignmentId: UUID): Future[Int]`

**Review by TomJKing — COMMENTED** [review]
Looks good. Just some minor comments. Thanks

File: src/main/scala/uk/gov/nationalarchives/tdr/api/service/FileStatusService.scala (hunk 2 above)
Comment (MancunianSam) [reply src/main/scala/uk/gov/nationalarchives/tdr/api/service/FileStatusService.scala:29]: I've added some in, thanks.

Comment (MancunianSam) [reply src/main/scala/uk/gov/nationalarchives/tdr/api/service/FileStatusService.scala:29]: I've added some in, thanks.

File: src/main/scala/uk/gov/nationalarchives/tdr/api/service/ConsignmentService.scala (hunk 3 above)
Comment (MancunianSam) [reply src/main/scala/uk/gov/nationalarchives/tdr/api/service/ConsignmentService.scala:128]: I can't believe I missed a chance to go deleting large parts of the API code. I've removed them and their associated tests.

File: src/main/scala/uk/gov/nationalarchives/tdr/api/service/ConsignmentService.scala (hunk 1 above)
Comment (MancunianSam) [reply src/main/scala/uk/gov/nationalarchives/tdr/api/service/ConsignmentService.scala:125]: I've taken these out as well. The service is looking much tidier now.

**Review by TomJKing — APPROVED** [review]
Looks good. Thanks

**Pull request merged**